src/controller/CHIPPersistentStorageDelegate.h

```diff
@@ -79,6 +79,19 @@ class DLL_EXPORT PersistentStorageDelegate
      */
     virtual void GetKeyValue(const char * key) = 0;
 
+    /**
+     * @brief
+     *   Lookup the key and call delegate object with it's stringified value.
+     *   This is a synchronous API, where the value is returned via the output
+     *   buffer. This API should be used sparingly, since it may block for
+     *   some duration.
+     *
+     * @param[in]      key Key to lookup
+     * @param[out]     value Value for the key
+     * @param[in, out] size Input value buffer size, output length of value
```

Comment:
This should make it clearer that output is length of value, which might be bigger than the input buffer size.  And that if that's the case, the value did not fit into the buffer and has been truncated.

But really, wouldn't it make more sense to have the API be that if we can't fit in the buffer we return failure?  Or is the idea that people should be able to interrogate the buffer size they need?  If so, I would rather we provided them with an explicit API for that....

Reply:
I added the documentation. The idea is that if the caller provides the correct sized buffer, the API will succeed. If not, instead of just returning the error, it also helps the caller by returning the buffer size needed. A few Posix style APIs provide similar facility. @rwalker-apple had proposed such style API in a previous code review.